model/context.go

```diff
@@ -396,3 +403,28 @@ func (s *Socket) fields() common.MapStr {
 	utility.Set(fields, "remote_address", s.RemoteAddress)
 	return fields
 }
+
+func clientFields(m common.MapStr, inpIP interface{}) common.MapStr {
+	ip := parseIP(inpIP)
+	if ip == "" {
+		m.Delete("ip")
+		return m
+	}
+	m["ip"] = ip
+	return m
+}
+
+func parseIP(i interface{}) string {
+	input, ok := i.(string)
+	if !ok {
+		return ""
+	}
+	var ip net.IP
+	if ip = net.ParseIP(input); ip != nil {
```

Comment:
Is a [net.SplitHostPort](https://golang.org/pkg/net/#SplitHostPort) on all input before parsing an option instead?

Reply:
Not on all input, as it expects a port to be present, but I can definitely use it for the second part, thank you.